--- configure.ac	2014-10-12 17:06:50.000000000 +0200
+++ configure.ac.new	2014-11-07 14:24:40.187700412 +0100
@@ -723,7 +723,13 @@
 AC_SUBST(TS_MAN8_MANPAGES)
 
 AC_MSG_CHECKING([whether to build man pages])
-AS_IF([test "x$sphinx_version_check" = "xyes" -a "x$SPHINXBUILD" != "xfalse"], [
+AC_ARG_ENABLE([man-pages],
+  [AS_HELP_STRING([--enable-man-pages],[build man pages])],
+  [],
+  [enable_man_pages="yes"]
+)
+
+AS_IF([test "x$sphinx_version_check" = "xyes" -a "x$SPHINXBUILD" != "xfalse" -a "xyes" = "x$enable_man_pages"], [
   AM_CONDITIONAL([BUILD_MANPAGES], [true])
   AC_MSG_RESULT([yes])
 ], [
